File: api/src/main/java/net/blueshell/api/repository/EventSignUpRepository.java
package net.blueshell.api.repository;

import net.blueshell.api.base.BaseRepository;
import net.blueshell.api.model.event.EventSignUp;
import net.blueshell.api.model.User;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface EventSignUpRepository extends BaseRepository<EventSignUp> {

    Optional<EventSignUp> findByUserIdAndEventId(Long userId, Long event);

    @Query("SELECT es FROM EventSignUp es WHERE es.guest.accessToken = :accessToken")
    Optional<EventSignUp> findByGuestAccessToken(@Param("accessToken") String accessToken);

    List<EventSignUp> findByUserId(Long user);

    List<EventSignUp> findByEventId(Long eventId);

    User user(User user);
}

File: api/src/main/kotlin/net/blueshell/api/repository/event/EventSignUpRepository.kt
package net.blueshell.api.repository.event

import net.blueshell.api.base.BaseRepository
import net.blueshell.api.model.event.EventSignUp
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.domain.Specification
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface EventSignUpRepository : BaseRepository<EventSignUp, Long> {
    @EntityGraph(value = "EventSignUp.withGuestAndAnswers", type = EntityGraph.EntityGraphType.LOAD)
    override fun findAll(spec: Specification<EventSignUp>): MutableList<EventSignUp>

    @EntityGraph(value = "EventSignUp.withGuestAndAnswers", type = EntityGraph.EntityGraphType.LOAD)
    override fun findAll(spec: Specification<EventSignUp>, pageable: Pageable): Page<EventSignUp>

    @EntityGraph(value = "EventSignUp.withGuestAndAnswers", type = EntityGraph.EntityGraphType.LOAD)
    fun findByUserIdAndEventId(userId: Long, eventId: Long): Optional<EventSignUp>

    @EntityGraph(value = "EventSignUp.withGuestAndAnswers", type = EntityGraph.EntityGraphType.LOAD)
    @Query("SELECT es FROM EventSignUp es WHERE es.guest.accessToken = :accessToken")
    fun findByGuestAccessToken(@Param("accessToken") accessToken: String): MutableList<EventSignUp>

    @EntityGraph(value = "EventSignUp.withGuestAndAnswers", type = EntityGraph.EntityGraphType.LOAD)
    fun findByEventId(eventId: Long): MutableList<EventSignUp>

    @EntityGraph(value = "EventSignUp.withGuestAndAnswers", type = EntityGraph.EntityGraphType.LOAD)
    fun findByGuestAccessTokenAndEventId(accessToken: String, eventId: Long): Optional<EventSignUp>

    @EntityGraph(value = "EventSignUp.withGuestAndAnswers", type = EntityGraph.EntityGraphType.LOAD)
    fun findAllByEventSignUpFormId(surveyId: Long): MutableSet<EventSignUp>
}
